compiler/ml/typetexp.ml

```diff
@@ -909,18 +911,49 @@ let report_error env ppf = function
        = Bar@}.@]@]"
       Printtyp.longident lid Printtyp.longident lid Printtyp.longident lid;
     spellcheck ppf fold_constructors env lid
-  | Unbound_label lid ->
+  | Unbound_label (lid, from_type) ->
     (* modified *)
-    Format.fprintf ppf
-      "@[<v>@{<info>%a@} refers to a record field, but no corresponding record \
-       type is in scope.@,\
-       @,\
-       If it's defined in another module or file, bring it into scope by:@,\
-       @[- Prefixing the field name with the module name:@ \
-       @{<info>TheModule.%a@}@]@,\
-       @[- Or specifying the record type explicitly:@ @{<info>let theValue: \
-       TheModule.theType = {%a: VALUE}@}@]@]"
-      Printtyp.longident lid Printtyp.longident lid Printtyp.longident lid;
+    (match from_type with
+    | Some {desc = Tconstr (p, _, _)} when Path.same p Predef.path_option ->
+      (* TODO: Extend for nullable/null? *)
+      Format.fprintf ppf
+        "@[<v>You're trying to access the record field @{<info>%a@}, but the \
+         thing you're trying to access it on is an @{<info>option@}.@ You need \
```

Comment:
Maybe 'value' instead of 'thing'?

Reply:
Done in https://github.com/rescript-lang/rescript/pull/7732/commits/ed7c2fb7287ea15a9bb35fd45cbc6eff72193984. Any other feedback?